aiida/engine/processes/functions.py

```diff
@@ -107,17 +107,13 @@ def run_get_node(*args, **kwargs):
             """
             Run the FunctionProcess with the supplied inputs in a local runner.
 
-            The function will have to create a new runner for the FunctionProcess instead of using the global runner,
-            because otherwise if this process function were to call another one from within its scope, that would use
-            the same runner and it would be blocking the event loop from continuing.
-
             :param args: input arguments to construct the FunctionProcess
             :param kwargs: input keyword arguments to construct the FunctionProcess
             :return: tuple of the outputs of the process and the process node pk
             :rtype: (dict, int)
             """
             manager = get_manager()
-            runner = manager.create_runner(with_persistence=False)
+            runner = manager.get_runner(with_persistence=False)
```

Comment:
The following docstring is there before I make the changes. This is what I mean 'blocking issues', I guess this was why a new event loop created in the ProcessFunction.

https://github.com/aiidateam/aiida-core/blob/5977dacc6160a68bba6fc4fd91c4bfa17f330520/aiida/engine/processes/functions.py#L110-L112

Reply:
> I'm not sure which 'blocking issues' you're referring to @unkcpz but indeed we now only have the one event loop so never need to create more than one. Does `get_runner` work if there is no event loop active at all or would this code be unreachable if that were the case?

`Manager.get_runner` will create a new runner if it doesn't have one yet. It won't pass in a loop to the `Runner` constructor, so that one will get whatever loop the constructor logic defines.